### PR TITLE
Add warning about VPC usage charges in automation

### PR DESCRIPTION
**Pull request by dreambeyondorange**
### Description of changes
* Adding a message to inform users that NAT Gateways incur charges outside of the free tier.

### Tests
* Tested the CLI locally to print the message during configure

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by himani2411 — DISMISSED** [review]

File: cli/src/pcluster/cli/commands/configure/easyconfig.py [hunk 1]
```diff
@@ -341,7 +341,18 @@ def _create_vpc_parameters(scheduler, head_node_instance_type, compute_instance_
     vpc_list = vpc_and_subnets["vpc_list"]
     if not vpc_list:
         print("There are no VPC for the given region. Starting automatic creation of VPC and subnets...")
-    if not vpc_list or prompt("Automate VPC creation? (y/n)", lambda x: x in ("y", "n"), default_value="n") == "y":
+    if (
+        not vpc_list
+        or prompt(
+            "The creation of the VPC will result in minimal charges for NAT gateway \n"
```
Comment (davprat) [inline cli/src/pcluster/cli/commands/configure/easyconfig.py:347]: Would it be better to say `standard` rather than `minimal` since minimal is subjective?

**Review by davprat — APPROVED** [review]
lgtm

File: cli/src/pcluster/cli/commands/configure/easyconfig.py (hunk 1 above)
Comment (dreambeyondorange) [reply cli/src/pcluster/cli/commands/configure/easyconfig.py:347]: That could be, but I don't want to alarm people with warning.  A NAT gateway cost ~$1 a day

File: cli/src/pcluster/cli/commands/configure/easyconfig.py [hunk 2]
```diff
@@ -341,7 +341,18 @@ def _create_vpc_parameters(scheduler, head_node_instance_type, compute_instance_
...
+            "creation using a public subnet that are not covered under the free tier.\n"
```
Comment (enrico-usai) [inline cli/src/pcluster/cli/commands/configure/easyconfig.py:348]: The warning message we're adding is about the "pcluster configure" command, so we need to take into consideration only two cases: public and public-private.
The NAT Gateway is created only in the public-private case, see templates used by pcluster configure here: https://github.com/aws/aws-parallelcluster/tree/develop/cloudformation/networking

So the message should be modified removing the "using a public subnet" sentence.


Then the NAT Gateway is not created at VPC creation time but at subnet creation time, so should be moved in the 
``` 
            or (prompt("Automate Subnet creation? (y/n)", lambda x: x in ("y", "n"), default_value="y") == "y")
```
only when selecting the public-private case.

**Pull request merged**